mono/mini/seq-points.c

```diff
@@ -11,31 +11,89 @@
 #include "seq-points.h"
 
 static void
-collect_pred_seq_points (MonoBasicBlock *bb, MonoInst *ins, GSList **next, int depth)
+insert_pred_seq_point (MonoInst *last_seq_ins, MonoInst *ins, GSList **next)
 {
-	int i;
 	MonoBasicBlock *in_bb;
 	GSList *l;
+	int src_index = last_seq_ins->backend.size;
+	int dst_index = ins->backend.size;
+
+	/* bb->in_bb might contain duplicates */
+	for (l = next [src_index]; l; l = l->next)
```

Comment:
N^2 where N is the number of incoming basic blocks to a basic block. This is bounded fairly tightly. If it's ever a problem, using a hash set would be a fairly trivial fix.